src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -88,16 +201,17 @@ object RegressionUtils {
 // constructs SparseVector of genotype calls with missing values mean-imputed
 // if all genotypes are missing then all elements are NaN
 class SparseGtBuilder extends Serializable {
-  private val missingRowIndices = new mutable.ArrayBuilder.ofInt()
-  private val rowsX = new mutable.ArrayBuilder.ofInt()
-  private val valsX = new mutable.ArrayBuilder.ofDouble()
+  private val missingRowIndices = new IntArrayBuilder()
+  private val rowsX = new IntArrayBuilder()
+  private val valsX = new DoubleArrayBuilder()
   private var row = 0
-  private var sparseLength = 0 // current length of rowsX and valsX, used to track missingRowIndices
+  private var sparseLength = 0
+  // current length of rowsX and valsX, used to track missingRowIndices
   private var nHet = 0
   private var nHomVar = 0
 
-  def merge(g: Genotype): SparseGtBuilder = {
-    (g.unboxedGT: @unchecked) match {
+  def merge(gt: Int): SparseGtBuilder = {
```

Comment:
I removed the split checks inside the methods because Tim added these `requireSplit()`.  But now I realize these are only checked when called through the "Scala API", not when LinearRegression is called directly.  I'll put them back in.

  def linreg(ySA: String, covSA: Array[String], root: String, minAC: Int, minAF: Double): VariantDataset = {
    requireSplit("linear regression")
    LinearRegression(vds, ySA, covSA, root, minAC, minAF)
  }